taf/oll/GitRepository.py

```diff
@@ -0,0 +1,59 @@
+import os
+import json
+from .utils import run
+
+
+class GitRepository(object):
```

Comment:
I would call this class `TAFTarget`, because it represents a target from `targets.json` file, with other information like `additional_info`, `repo_urls`,...
Then we could have a lazy loaded [GitRepo](https://github.com/openlawlibrary/taf/blob/bee39746066da25423227f3628f4740cdd62fbcb/taf/oll/updater/git.py#L29) property, if someone wants to call a git functions for that target.

Reply:
I thought about both ideas before, but I decided to name and implement `GitRepository` like this for a few reasons. We  instantiate the repositories based on `repositories.json`, meaning that TAF can instantiate a repository which is not specified in `targets.json`. This is important to us because we need to handle preview repositories, but I think that it's a generally good idea to support that. Since TAF is designed to handle git repositories, I think that having a class named `GitRepository` makes sense.
I also thought about having a Git repository as a property, but then we would have to call `git_repo.some_git_method` every time instead of just `some_fit_method`. Since the class is supposed to represent a Git repository anyway, I think that it makes sense to put all methods directly in it. I though about `additional_info`, and at first I didn't like storing such data inside `GitRepository`. However, since that additional info is specified in `repositories.json` or `targets.json` and is directly related to the repository, I prefer the final solution.

Reply:
I am fine with that, just this class seemed too empty for me. I would suggest just adding some basic git functionalities to this class then, what do you think?

Reply:
I added just what we need in TAF. Do you think it would make sense to add functionalities like commit, push, pull, create branch and such?

Reply:
I think yes, and if someone wants some specific methods, custom subclass can be provided. But for me it makes sense to have at least `git pull`.

Reply:
I added moved a few more methods from platform. I think that, since TAF has functionalities for validating branches, it makes sense to add functionalities for creating and checking out branches. Also added basics like commit, push and pull